### ISSUE TITLE
PoET for peptide binding affinity prediction

### ISSUE THREAD

**Issue by vladimirkovacevic** [OpenProteinAI/PoET#2]
Would it make sense to utilise PoET for binding affinity prediction (ic50 score) of short peptide sequences (8-16 amino acids)? If yes, do you have any suggestions/comments on how to do it? Is it possible to obtain embedding (representation) for each protein sequence and pass it to some custom classification head that will be trained?

Thanks!

**Comment by timt51** [comment]
We haven't really tested with short peptides in particular, but if you have at least a couple of relevant homologs to condition on, PoET should be helpful. Fine-tuning PoET or training a model on top of PoET embeddings as you suggest are definitely reasonable ways of utilizing PoET for this!

The [embed method](https://github.com/OpenProteinAI/PoET/blob/9b2239be84ee39691ec6ad4184925156f2ac332f/poet/models/poet.py#L731) (example usage in [scoring script](https://github.com/OpenProteinAI/PoET/blob/9b2239be84ee39691ec6ad4184925156f2ac332f/scripts/score.py#L175)) can be used to get embeddings. If you want the embeddings for a lot of sequences conditioned on the same prompt, you can do so more efficiently using the [logits method](https://github.com/OpenProteinAI/PoET/blob/main/poet/models/poet.py#L848) with `return_embeddings=True` (example usage in [scoring script](https://github.com/OpenProteinAI/PoET/blob/9b2239be84ee39691ec6ad4184925156f2ac332f/scripts/score.py#L155)).

If you're interested, we also offer [REST](https://docs.openprotein.ai/rest-api/embeddings.html#/) and [Python](https://docs.openprotein.ai/python-api/poet/scoring-sequences.html) APIs to get embeddings from PoET, which you can request access to through [this form](https://www.openprotein.ai/early-access-form) and is free for academic use.

**Comment by vladimirkovacevic** [comment]
Thank you @timt51 for very detailed explanation. I'll try to test what you've suggested. Also, I've filled the form you linked. I'll let you know how it went.
